exporter/containerimage/writer.go

```diff
@@ -409,6 +409,32 @@ func normalizeLayersAndHistory(diffs []blobs.DiffPair, history []ocispec.History
 		history[i] = h
 	}
 
+	var created *time.Time
+	var noCreatedTime bool
+	for _, h := range history {
+		if h.Created != nil {
+			created = h.Created
+			if noCreatedTime {
+				continue
+			}
```

Comment:
Its supposed to be `break`

Reply:
Then is `noCreatedTime` useful in this variant of the loop, isn't an unconditional loop fine?

Reply:
We want to break after first nil record. Otherwise, the history item for a first metadata command would be the creation time of a base image layer.

Reply:
So you are looking for the second layer with a timestamp (because the first is always a base layer)? If so I think it'd be worth expanding the comment.

Reply:
First layer with a timestamp after a record without a timestamp. If there is no such then the last layer with timestamp.

Reply:
@tonistiigi I think it would be very useful to sum up this conversation in a comment, because I don't think this is obvious at all.

Reply:
(probably doesn't need saying, but I agree with @tiborvass ...)